$MirOS: ports/devel/glib/patches/patch-glib_gconvert_c,v 1.2 2008/03/06 22:08:24 tg Exp $
--- glib/gconvert.c.orig	Fri Sep 12 19:55:28 2008
+++ glib/gconvert.c	Thu Nov 25 12:24:44 2010
@@ -48,13 +48,6 @@
 
 #include "glibintl.h"
 
-#if defined(USE_LIBICONV_GNU) && !defined (_LIBICONV_H)
-#error GNU libiconv in use but included iconv.h not from libiconv
-#endif
-#if !defined(USE_LIBICONV_GNU) && defined (_LIBICONV_H)
-#error GNU libiconv not in use but included iconv.h is from libiconv
-#endif
-
 #include "galias.h"
 
 GQuark 
@@ -735,6 +728,9 @@ g_convert (const gchar *str,
   g_return_val_if_fail (to_codeset != NULL, NULL);
   g_return_val_if_fail (from_codeset != NULL, NULL);
   
+  if (strcmp (to_codeset, from_codeset) == 0)
+    return g_strndup (str, len);
+
   cd = open_converter (to_codeset, from_codeset, error);
 
   if (cd == (GIConv) -1)
